compiler/rustc_borrowck/src/diagnostics/mutability_errors.rs

```diff
@@ -1216,6 +1220,56 @@ impl<'a, 'tcx> MirBorrowckCtxt<'a, 'tcx> {
                 if let Some(hir_id) = hir_id
                 && let Some(hir::Node::Local(local)) = hir_map.find(hir_id)
                 {
+
+                    let initial_expression = match local.init.map(|init| init.kind)  {
+                        Some(hir::ExprKind::Path(hir::QPath::Resolved(_, hir::Path{res: Res::Local(id), ..} ))) => hir_map.find(*id),
+                        Some(hir::ExprKind::Call(Expr {  kind: ExprKind::Path(QPath::Resolved(_, Path {res: Res::Def(DefKind::Fn, defid ) , ..}   ), ..), ..}, ..  )) => {
+                            let Some(local_id) = defid.as_local() else {
+                                todo!("What do we do here?")
```

Comment:
We *could* have a fallback case that just has a more lax "you may be able to change the mutability here". I'm not sure if the binding info is available cross-crates.